### ISSUE TITLE
make install segfaults

### ISSUE THREAD

**Issue by mnieper** [ashinn/chibi-scheme#394]
I just pulled the newest sources. `sudo make install` ends with:

> cd / && LD_LIBRARY_PATH="/usr/local/lib:" DYLD_LIBRARY_PATH="/usr/local/lib:" /usr/local/bin/chibi-scheme -mchibi.snow.commands -mchibi.snow.interface -mchibi.snow.package -mchibi.snow.utils -d /usr/local/share/chibi/snow.img
Segmentation fault (core dumped)

This may, or may not be related. GCC now warns about unsafe conversions:

```
cc -fPIC -shared -DSEXP_USE_NTPGETTIME -DSEXP_USE_INTTYPES -Iinclude  -Wall -g -g3 -O3   -o lib/srfi/27/rand.so lib/srfi/27/rand.c -L.  -lchibi-scheme
In file included from include/chibi/eval.h:12:0,
                 from lib/srfi/27/rand.c:6:
lib/srfi/27/rand.c: In function ‘sexp_make_random_source’:
include/chibi/sexp.h:1105:36: warning: passing argument 3 of ‘sexp_alloc_tagged_aux’ makes integer from pointer without a cast [-Wint-conversion]
 #define sexp_opcode_return_type(x) (sexp_field(x, opcode, SEXP_OPCODE, ret_type))
                                    ^
include/chibi/sexp.h:655:76: note: in definition of macro ‘sexp_alloc_tagged’
  sexp_alloc_tagged(ctx, size, tag) sexp_alloc_tagged_aux(ctx, size, tag sexp_with_current_source(__FILE__, __LINE__))
                                                                     ^~~
lib/srfi/27/rand.c:154:52: note: in expansion of macro ‘sexp_opcode_return_type’
   res = sexp_alloc_tagged(ctx, sexp_sizeof_random, sexp_opcode_return_type(self));
                                                    ^~~~~~~~~~~~~~~~~~~~~~~
In file included from include/chibi/eval.h:12:0,
                 from lib/srfi/27/rand.c:6:
include/chibi/sexp.h:1460:15: note: expected ‘sexp_uint_t {aka long unsigned int}’ but argument is of type ‘sexp {aka struct sexp_struct *}’
 SEXP_API sexp sexp_alloc_tagged_aux(sexp ctx, size_t size, sexp_uint_t tag sexp_current_source_param);
               ^~~~~~~~~~~~~~~~~~~~~
```

**Comment by mnieper** [comment]
```
==25571== Memcheck, a memory error detector
==25571== Copyright (C) 2002-2015, and GNU GPL'd, by Julian Seward et al.
==25571== Using Valgrind-3.12.0.SVN and LibVEX; rerun with -h for copyright info
==25571== Command: /usr/local/bin/chibi-scheme -mchibi.snow.commands -mchibi.snow.interface -mchibi.snow.package -mchibi.snow.utils -d /usr/local/share/chibi/snow.img
==25571== 
==25571== Invalid read of size 2
==25571==    at 0x4E44674: sexp_mark_one (gc.c:203)
==25571==    by 0x4E44717: sexp_mark_one (gc.c:212)
==25571==    by 0x4E44717: sexp_mark_one (gc.c:212)
==25571==    by 0x4E44717: sexp_mark_one (gc.c:212)
==25571==    by 0x4E44717: sexp_mark_one (gc.c:212)
==25571==    by 0x4E44717: sexp_mark_one (gc.c:212)
==25571==    by 0x4E44717: sexp_mark_one (gc.c:212)
==25571==    by 0x4E44717: sexp_mark_one (gc.c:212)
==25571==    by 0x4E44717: sexp_mark_one (gc.c:212)
==25571==    by 0x4E44717: sexp_mark_one (gc.c:212)
==25571==    by 0x4E44717: sexp_mark_one (gc.c:212)
==25571==    by 0x4E44717: sexp_mark_one (gc.c:212)
==25571==  Address 0x50 is not stack'd, malloc'd or (recently) free'd
==25571== 
==25571== 
==25571== Process terminating with default action of signal 11 (SIGSEGV)
==25571==  Access not within mapped region at address 0x50
==25571==    at 0x4E44674: sexp_mark_one (gc.c:203)
==25571==    by 0x4E44717: sexp_mark_one (gc.c:212)
==25571==    by 0x4E44717: sexp_mark_one (gc.c:212)
==25571==    by 0x4E44717: sexp_mark_one (gc.c:212)
==25571==    by 0x4E44717: sexp_mark_one (gc.c:212)
==25571==    by 0x4E44717: sexp_mark_one (gc.c:212)
==25571==    by 0x4E44717: sexp_mark_one (gc.c:212)
==25571==    by 0x4E44717: sexp_mark_one (gc.c:212)
==25571==    by 0x4E44717: sexp_mark_one (gc.c:212)
==25571==    by 0x4E44717: sexp_mark_one (gc.c:212)
==25571==    by 0x4E44717: sexp_mark_one (gc.c:212)
==25571==    by 0x4E44717: sexp_mark_one (gc.c:212)
==25571==  If you believe this happened as a result of a stack
==25571==  overflow in your program's main thread (unlikely but
==25571==  possible), you can try to increase the size of the
==25571==  main thread stack using the --main-stacksize= flag.
==25571==  The main thread stack size used in this run was 8388608.
==25571== 
==25571== HEAP SUMMARY:
==25571==     in use at exit: 6,310,149 bytes in 56 blocks
==25571==   total heap usage: 520 allocs, 464 frees, 6,691,080 bytes allocated
==25571== 
==25571== LEAK SUMMARY:
==25571==    definitely lost: 0 bytes in 0 blocks
==25571==    indirectly lost: 0 bytes in 0 blocks
==25571==      possibly lost: 0 bytes in 0 blocks
==25571==    still reachable: 6,310,149 bytes in 56 blocks
==25571==         suppressed: 0 bytes in 0 blocks
==25571== Rerun with --leak-check=full to see details of leaked memory
==25571== 
==25571== For counts of detected and suppressed errors, rerun with: -v
==25571== ERROR SUMMARY: 1 errors from 1 contexts (suppressed: 0 from 0)
Speicherzugriffsfehler (Speicherabzug geschrieben)
```

**Comment by ashinn** [comment]
Careless fix from issue #385, forgot to test on Linux.  Compiles and installs fine now but I want to double check image support on Linux in general.
